apps/server/src/modules/pseudonym/repo/external-tool-pseudonym.repo.ts

```diff
@@ -71,15 +75,33 @@ export class ExternalToolPseudonymRepo {
 		return promise;
 	}
 
+	async findPseudonymByPseudonym(pseudonym: string): Promise<Pseudonym | null> {
+		const entities: ExternalToolPseudonymEntity[] = await this.em.find(ExternalToolPseudonymEntity, { pseudonym });
```

Comment:
you can use findOne, since pseudonym are expected to be unique (see entity definition)